src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/MessageBody.cs

```diff
@@ -217,6 +217,58 @@ private void ConsumedBytes(int count)
             OnConsumedBytes(count);
         }
 
+        internal static unsafe void ParseConnection(string connection, out bool isKeepAlive, out bool isUpgrade)
+        {
+            isKeepAlive = false;
+            isUpgrade = false;
+
+            fixed (char* ptr = connection)
+            {
+                var ch = ptr;
+                var end = ch + connection.Length;
+
+                while (ch < end)
+                {
+                    if ((*ch | 0x20) == 'k' && (end - ch) >= 9)
+                    {
+                        if ((*++ch | 0x20) == 'e' &&
```

Comment:
For ulong compare you want to use the magic number ;-) Gowan....

https://github.com/davidfowl/Channels/pull/120